### PR TITLE
integration: don't use low-level etcd master setup for dry-run+admission-webhook tests

### PR DESCRIPTION
**Pull request by sttts**
We should use standard plumbing to start up integration test components. Neither dry-run nor admission-webhooks need special low-level settings.

### CODE REVIEW

File: test/integration/apiserver/admissionwebhook/admission_test.go [hunk 1]
```diff
@@ -323,25 +325,20 @@ func TestWebhookV1beta1(t *testing.T) {
...
-		// force enable all resources so we can check storage.
-		// TODO: drop these once we stop allowing them to be served.
-		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/deployments"] = "true"
-		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/daemonsets"] = "true"
-		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/replicasets"] = "true"
-		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/podsecuritypolicies"] = "true"
-		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/networkpolicies"] = "true"
-	})
-	defer master.Cleanup()
+	s, err := kubeapiservertesting.StartTestServer(t, kubeapiservertesting.NewDefaultTestServerOptions(), []string{
```
Comment (liggitt) [inline test/integration/apiserver/admissionwebhook/admission_test.go]: StartRealMasterOrDie also sets up test CRDs:

```go
	// create CRDs so we can make sure that custom resources do not get lost
	CreateTestCRDs(t, apiextensionsclientset.NewForConfigOrDie(kubeClientConfig), false, GetCustomResourceDefinitionData()...)
```

Comment (sttts) [reply test/integration/apiserver/admissionwebhook/admission_test.go]: what for?

Comment (sttts) [reply test/integration/apiserver/admissionwebhook/admission_test.go]: I would prefer standard plumbing and explicit fixtures than something hidden deep in custom plumbing code.

Comment (liggitt) [reply test/integration/apiserver/admissionwebhook/admission_test.go]: no disagreement, we can hoist that fixture setup out to the places that require it. it grew organically and all the tests using that setup method intend to exercise an operation (checking etcd storage path, dry-run, or webhook admission) over all resources, including custom resources

Comment (sttts) [reply test/integration/apiserver/admissionwebhook/admission_test.go]: fixed

File: test/integration/apiserver/admissionwebhook/admission_test.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
 	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	apiexternsionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
```
Comment (liggitt) [inline test/integration/apiserver/admissionwebhook/admission_test.go]: nit: typo "apiexternsionsclientset"

**Comment by liggitt** [comment]
/lgtm
/approve

/hold
for nit

**Comment by sttts** [comment]
fix nit

**Pull request merged**